scipy/integrate/_ivp/ivp.py

```diff
@@ -652,7 +652,15 @@ def fun(t, x, fun=fun):
 
     status = None
     while status is None:
-        message = solver.step()
+        try:
+            message = solver.step()
+        except Exception as e:
+            if options.get('catch_exceptions', False):
```

Comment:
The option, the new status and the whole logic needs to be documented.

I don't see significant downsides if keeping this logic optional.

Reply:
Thanks for the review! I will work on some documentation for this option.